### PR TITLE
Update observe modules

### PR DESCRIPTION
**Pull request by andrea-prearo**
This PR addresses Issue #18.

### CODE REVIEW

**Comment by andrea-prearo** [comment]
Hey @Geek-1001,

I had to migrate more files than initially planned. In particular, after migrating `CLUObserveModule` (as `ObserveModule`), I ran into the issue on not being able to subclass the Swift class in Objective C. Solving that, required to migrate pretty much all the files in `Modules` to Swift.

File: Clue/Classes/Modules/UserInteractionModule.swift [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+//
+//  Created by Andrea Prearo on 5/8/17.
+//  Copyright © 2017 Ahmed Sulaiman. All rights reserved.
+//
+
+import Foundation
+
+let UserInteractionTouchBeganEvent = "Began"
+let UserInteractionTouchMovedEvent = "Moved"
+let UserInteractionTouchEndedEvent = "Ended"
```
Comment (Geek-1001) [inline Clue/Classes/Modules/UserInteractionModule.swift:13]: What do you think about changing those constants into enum type with `String` as a raw type?
Something like this:
```swift
enum UserInteractionEventType: String {
    case began = "Began"
    case moved = "Moved"
    case ended = "Ended"
}
```

File: Clue/Classes/Modules/DeviceInfoModule.swift [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        let deviceProperties = getDeviceProperties()
+        writer.startWriting()
+        writer.append(json: deviceProperties)
+        writer.finishWriting()
+    }
+}
+
+// MARK: - Private Methods
+fileprivate extension DeviceInfoModule {
+    func getDeviceProperties() -> [AnyHashable: Any] {
```
Comment (Geek-1001) [inline Clue/Classes/Modules/DeviceInfoModule.swift:31]: What do you think about encapsulation of this function into the separate `UIDevice` extension? 
Something like we did with `UIView` category to parse its properties for example.
```swift
let device = UIDevice.current
var deviceProperties = device.clueProperties() // which will return dictionary with properties
```

**Comment by Geek-1001** [comment]
Hello @andrea-prearo 
Everything is awesome. Thank you so much for your contribution! 🤘

I have a few comment in some places. I'd like to wait for your thoughts on those.
And I'll merge this PR.

File: Clue/Classes/Modules/UserInteractionModule.swift (hunk 1 above)
Comment (andrea-prearo) [reply Clue/Classes/Modules/UserInteractionModule.swift:13]: Sounds good to me.

File: Clue/Classes/Modules/DeviceInfoModule.swift (hunk 2 above)
Comment (andrea-prearo) [reply Clue/Classes/Modules/DeviceInfoModule.swift:31]: Sounds good to me.

**Comment by andrea-prearo** [comment]
Thanks @Geek-1001!

As the migration work required a lot of changes, I was thinking about postponing some of the refactoring (like the changes you suggested) to a subsequent PR. Just to avoid having really big PRs.

We could create a new issue, just to keep track of our refactoring goals, and I can take care of that and implement all the suggested changes in a new PR.
What do you think about that?

**Comment by Geek-1001** [comment]
Hi @andrea-prearo 
Yes, totally make sense. I'll add some issues regarding refactoring later today.

I'm merging this PR.

**Pull request merged**